docs/site/Context.md

```diff
@@ -372,7 +388,31 @@ app
 // bind: foo-app
 ```
 
-It's recommended that `ContextObserver` implementations should not throw errors
-in `observe` method. Uncaught errors will be reported as `error` events of the
-context object. If there is no error listener registered, the node process will
-crash.
+Please note when an observer subscribes to a context, it will be registered with
+all contexts on the chain. In the example above, the observer is added to both
+`server` and `app` contexts so that it can be notified when bindings are added
+or removed from any of the context on the chain.
+
+- Observers are called in the next turn of
+  [Promise micro-task queue](https://jsblog.insiderattack.net/promises-next-ticks-and-immediates-nodejs-event-loop-part-3-9226cbe7a6aa)
+
+- When there are multiple async observers registered, they are notified in
+  series for an event.
+
+- When multiple binding events are emitted in the same event loop tick and there
+  are async observers registered, such events are queued and observers are
+  notified by the order of events.
+
+### Observer error handling
+
+It's recommended that `ContextEventObserver` implementations should not throw
+errors in their code. Errors thrown by context event observers are reported as
+follows over the context chain.
+
+1. Check if the current context object has `error` listeners, if yes, emit an
+   `error` event on the context and we're done. if not, try its parent context
+   by repeating step 1.
+
+2. If no context object of the chain has `error` listeners, emit an `error`
+   event on the current context. As a result, the process exits abnormally. See
```

Comment:
Personally, I would choose a slightly different behavior: _If no context object of the chain has `error` listeners, emit an `error` event on the root context  (typically the Application class)._

I.e. let the error bubble up the context chain util we either find a context with error handlers or reach the top.

Just an idea to consider.

Reply:
I leave it to the current context so that we know which one causes the crash. The end result should be the same.